io/pci/distro_tools.py

```diff
@@ -30,27 +30,61 @@ class DisrtoTool(Test):
     '''
     to test different type of tool
     '''
-    @skipUnless(IS_POWER_VM,
-                "supported only on PowerVM platform")
+
     def setUp(self):
         '''
-        to check host platform
+        get all parameters
         '''
-        self.option = self.params.get("option", default='')
+        self.option = self.params.get("test_opt", default='').split(",")
+        self.tool = self.params.get("tool", default='')
 
-    def test(self):
+    @skipUnless(IS_POWER_VM,
+                "supported only on PowerVM platform")
+    def lsslot(self, option):
         '''
-        test the lsslot tool
+        run lsslot
         '''
         cmd = "lsslot"
-        if self.option == "pci":
-            cmd = "%s -d %s" % (cmd, self.option)
+        if option == "pci":
+            cmd = "%s -d %s" % (cmd, option)
         else:
-            cmd = "%s -c %s" % (cmd, self.option)
+            cmd = "%s -c %s" % (cmd, option)
         result = process.run(cmd, shell=True, ignore_status=True)
         if result.exit_status != 0:
             self.fail("Failed to display hot plug slots")
 
```

Comment:
we can make this check more robust by checking the existence of our IO adapter in result instead of just checking the exist_status.